save-plugins/warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugin/warn/utils/Warning.kt

```diff
@@ -0,0 +1,46 @@
+/**
+ * Classes and methods to work with warnings
+ */
+
+package org.cqfn.save.plugin.warn.utils
+
+/**
+ * Class for warnings which should be discovered and compared wit analyzer output
+ *
+ * @property message warning text
+ * @property line line on which this warning occurred
+ * @property column column in which this warning occurred
+ */
+data class Warning(
+    val message: String,
+    val line: Int?,
+    val column: Int?,
+)
+
+/**
+ * Extract warning from [this] string using provided parameters
+ *
+ * @param warningRegex regular expression for warning
+ * @param columnGroupIdx index of capture group for column number
+ * @param lineGroupIdx index of capture group for line number
+ * @param messageGroupIdx index of capture group for waring text
+ * @return a [Warning] or null if [this] string doesn't match [warningRegex]
+ */
+internal fun String.extractWarning(warningRegex: Regex,
+                                   columnGroupIdx: Int?,
+                                   lineGroupIdx: Int?,
+                                   messageGroupIdx: Int,
+): Warning? {
+    val groups = warningRegex.find(this)?.groups ?: return null
+    val line = lineGroupIdx?.let {
+        groups[lineGroupIdx]?.value?.toInt()
+    }
+    val column = columnGroupIdx?.let {
+        groups[columnGroupIdx]?.value?.toInt()
+    }
+    return Warning(
+        groups[messageGroupIdx]!!.value,
```

Comment:
and need some logging here (proper exception)

Reply:
I think this can be done in #51 as well